pkg/apis/triggers/v1alpha1/trigger_binding_types.go

```diff
@@ -0,0 +1,63 @@
+package v1alpha1
+
+/*
+Copyright 2019 The Tekton Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Check that TriggerBinding may be validated and defaulted.
+//var _ apis.Validatable = (*TriggerBinding)(nil)
+
+// +k8s:deepcopy-gen=true
+type TriggerBindingSpec struct {
+	Templates []TriggerTemplateRef `json:"templates,omitempty"`
+}
+
+// +k8s:deepcopy-gen=true
+type TriggerTemplateRef struct {
+	TemplateRef string  `json:"templateref,omitempty"`
```

Comment:
To me, the `TemplateRef` seems analogous to the previously defined [`TriggerBindingRef`](https://github.com/tektoncd/triggers/pull/17/files?file-filters%5B%5D=.c&file-filters%5B%5D=.db&file-filters%5B%5D=.go&file-filters%5B%5D=.libyaml&file-filters%5B%5D=.md&file-filters%5B%5D=.rc&file-filters%5B%5D=.s&file-filters%5B%5D=.sh&file-filters%5B%5D=No+extension#diff-e9c8d0e7ab4b5d2e8bbec5f6eafbb4a6R30), so I think it should probably have its own struct for the ref that includes a `Name`, `APIVersion`, and possibly `Namespace` (see my comment above about the `Namespace` 😁 ). But let me know if there's a separate motivation for keeping it as a `string` (I think it was a string in the design doc, but that might've been for brevity)

Reply:
Make sense to me! Updated